### ISSUE TITLE
`Refector` plugin doesn't strip GHC suggestion on `Define` code action

### ISSUE THREAD

**Issue by s-and-witch** [haskell/haskell-language-server#4165]
HLS version:

I'm using 
```
$ haskell-language-server --version 
haskell-language-server version: 2.6.0.0 (GHC: 9.6.4)
```
But this behavior could be reproduced with earlier versions of GHC.

### Steps to reproduce

Try to use `Define` code action on name that is close enough to already existing identifier, e.g.

```haskell
main :: IO ()
main = fooo ()

foo :: Int
foo = _
```
Here we already have `foo` (two "o") in the scope and on unknown identifier `fooo` (three "o") GHC will suggest fix "Perhaps use ‘foo’". This suggestion goes directly into code action, so it will define 
```haskell
fooo :: () -> IO () Suggested fix: Perhaps use ‘foo’ (line 7)
fooo = _
```
<details>
  <summary>Screenshots</summary>
  
![изображение](https://github.com/haskell/haskell-language-server/assets/74188979/1d083834-d4ae-44ef-8dcd-abd715b6aeeb)
![изображение](https://github.com/haskell/haskell-language-server/assets/74188979/8cc39b16-94e8-431c-a0f3-000c49ece307)

</details>

### Expected behaviour

Strip the suggestion
```haskell
fooo :: () -> IO ()
fooo = _
```

### Related code

As I can see, related code is this function: https://github.com/haskell/haskell-language-server/blob/1dd54a50495a3980988261250cdc97dcf718ed0c/plugins/hls-refactor-plugin/src/Development/IDE/Plugin/Plugins/Diagnostic.hs#L46-L52

It propagates parsed type to code action builder here: https://github.com/haskell/haskell-language-server/blob/1dd54a50495a3980988261250cdc97dcf718ed0c/plugins/hls-refactor-plugin/src/Development/IDE/Plugin/CodeAction.hs#L880-L882

And `newDefinitionAction` builds code action without any post-processing.

I'm not sure why `typ:_ <- T.splitOn " Suggested fix:" typ0` doesn't work, additional debug is required.

**Comment by michaelpj** [comment]
I _think_ this got fixed recently? https://github.com/haskell/haskell-language-server/pull/4080 ?

Might well not be in 2.6, can you try with a later version or master?

**Comment by jhrcek** [comment]
This should be fixed in 2.7.0.0.
I can't reproduce it with that anymore

![fixed](https://github.com/haskell/haskell-language-server/assets/2716069/2fb7b079-48de-4715-99f5-cffa75710165)

**Comment by fendor** [comment]
Thanks for verifying! Closing this as already fixed.
